src/rust-crypto/rust-crypto.ts

```diff
@@ -1041,7 +1041,8 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      */
     public async onLiveEventFromSync(event: MatrixEvent): Promise<void> {
         // Ignore state event or remote echo
-        if (event.isState() || event.getUnsigned().transaction_id === null) return;
+        // transaction_id is provided in case of remote echo {@link https://spec.matrix.org/v1.7/client-server-api/#local-echo}
+        if (event.isState() || "transaction_id" in event.getUnsigned()) return;
```

Comment:
although this is technically correct, in other places where we make this check, we look for truthiness of `transaction_id`. In other words:

```suggestion
        if (event.isState() || !!event.getUnsigned().transaction_id) return;
```

Let's be consistent.